[An Cheng Yang] iP

anchengyang opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes/))

Duke frees your mind by remembering things you need to do

It's 
- Text-based
- Easy to learn
- ~~Fast~~ _SUPER FAST_ to use

## Steps to run
1. Pull the code from the repo [here](https://github.com/anchengyang/ip)
2. Run it
3. Add your tasks
4. Let it manage your tasks for you 👍

## Features
- [x] Managing todo
- [x] Managing deadlines
- [x] Managing events
- [x] Filter by keyword
- [x] Mark and unmark tasks
- [x] Filter by date

This is the main function:
   ```ruby
   public static void main(String[] args) {
        new Duke("tasks.txt").run();
    }
   ```

Commit:
Add Gradle support

anchengyang committed:
Level 1

anchengyang committed:
Level 2

anchengyang committed:
Level 2

anchengyang committed:
Level-3

anchengyang committed:
Level-3

anchengyang committed:
Level-4

anchengyang committed:
TextUiTesting

anchengyang committed:
Level-5

anchengyang committed:
Level-5

anchengyang committed:
Level 5

anchengyang committed:
Added more exception handlers

anchengyang committed:
Level 6

anchengyang committed:
enums added

anchengyang committed:
Merge pull request #1 from anchengyang/testbed

merge Testbed branch

anchengyang committed:
minimal level 8

anchengyang committed:
Add save to hard disk & TaskList class

anchengyang committed:
Add Date and Time to Deadline

anchengyang committed:
Bug fixes for deadline date time

anchengyang committed:
More OOP

anchengyang committed:
Merge branch 'add-gradle-support'

anchengyang committed:
merge add-gradle-support branch

anchengyang committed:
Add junit tests

anchengyang committed:
build jar file

anchengyang committed:
Add Find command

anchengyang committed:
fix code style

anchengyang committed:
Add java docs

anchengyang committed:
Adhere to coding standards

anchengyang committed:
Add Find command

anchengyang committed:
Add find command to welcome message

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
checkstyle

anchengyang committed:
Merge branch 'master' of https://github.com/anchengyang/ip

anchengyang committed:
Fix import checkstyle

anchengyang committed:
Base javaFX application until step 3

anchengyang committed:
Part 4 of tutorial (FXML load error)

igezt commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,59 @@
+     * Exits from the while loop if the command is a ByeCommand.
+     */
+    public void run() {
+        this.ui.showWelcomeMessage();
```
Good job on abstracting away the welcome message into the UI!

igezt commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+        default:
+            throw new IllegalArgumentException();
+        }
+    }
```
Very clean parser code! Do you have an exception for if the input body is empty?

igezt commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,73 @@
+            oos.close();
+            fos.close();
+            System.out.println("    Saved");
+        } catch (Exception e) {
```
Maybe try not to shorten the variables name too much?

fos -> fileOutput
oos -> objectOutput

The original variable names are concise but don't really tell the other viewers what they are.

igezt commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,73 @@
+        if (file.exists()) {
+            TaskList tl = new TaskList();
+            try {
+                FileInputStream fis = new FileInputStream(this.filePath);
```
Same thing for here as above.

igezt commented on src/main/java/duke/commands/ByeCommand.java:
```diff
@@ -0,0 +1,37 @@
+        storage.saveTaskList(tasks);
+        System.out.println("    " + "Bye. Hope to see you again soon!");
+        return true;
+    }
```
Maybe you can pass the response text to the Ui object for them to settle instead!

igezt reviewed (commented):
I think your code looks good!! Maybe need to abstract out a few more exceptions to be more specific instead of just passing in the error message to the DukeException constructor. Also, your variables tend to not really inform the user what they are (especially in the case of your shortened variables). Typically, when working in a group, it is necessary to make it easy to understand at a glance what the variable is doing.

zuohui48 commented on src/main/java/duke/commands/Command.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * The abstract class for command.
+ */
+public abstract class Command {
```
Really good use of OOP to distinguish each command as a separate class!

zuohui48 commented on src/main/java/duke/commands/ByeCommand.java:
```diff
@@ -0,0 +1,37 @@
+     */
+    public boolean execute(TaskList tasks, Ui ui, Storage storage) {
+        storage.saveTaskList(tasks);
+        System.out.println("    " + "Bye. Hope to see you again soon!");
```
I agree with @igezt, the response text from Duke could be abstracted into the Ui object

zuohui48 reviewed (commented):
Really clean code! LGTM.

anchengyang commented:
thanks for the comments @igezt @zuohui48 i will work on them!

anchengyang committed:
resolve previous issue

anchengyang committed:
add style

anchengyang committed:
fix ui

anchengyang committed:
fix style error

anchengyang committed:
find date and abstract responses into ui

anchengyang committed:
Change mainClassName in gradle

anchengyang committed:
Add assertions to mark, unmark, list, delete and dates

anchengyang committed:
Fix grammar error in Event constructor comment

anchengyang committed:
Improve code quality

There are many error statements all over the command classes.

Unifying them under UI improves code quality.

As a step towards such unification, let's extract these statements
and separate them into methods within Ui class. Doing so will make
the subsequent unification easier.

anchengyang committed:
Merge pull request #2 from anchengyang/branch-A-Assertions

Branch a assertions

anchengyang committed:
Merge pull request #3 from anchengyang/branch-A-CodeQuality

Branch a code quality

anchengyang committed:
Use streams in find and finddate

Streams are optimized for filter functions.

Since filter functions are used to find by task description
and task dates, the for loops were replaced by streams to help
optimize the filter.

anchengyang committed:
Merge pull request #4 from anchengyang/branch-A-Streams

Use streams in find and finddate

anchengyang committed:
Update test files

anchengyang committed:
Create gradle.yml

anchengyang committed:
remove IO redirection test

anchengyang committed:
Update gradle.yml

anchengyang committed:
Update gradle.yml

anchengyang committed:
Add update description command

We did not have a way to edit the description of
tasks without deleting the task and creating a new task.

Now we are able to update the description using an update
command with a /description prefix.

anchengyang committed:
Update date for deadline tasks

anchengyang committed:
style check edits

anchengyang committed:
Merge pull request #5 from anchengyang/branch-C-Update

Branch c update

anchengyang committed:
Fix code quality

anchengyang committed:
small UI changes

anchengyang committed:
improved Gui

anchengyang committed:
checkstyle fixes

anchengyang committed:
Fix style

anchengyang committed:
Update existing unit tests

anchengyang committed:
Add find command unit test

anchengyang committed:
Remove assertions

anchengyang committed:
Delete command test

anchengyang committed:
Merge branch 'branch-A-MoreErrorHandling'

anchengyang committed:
Add more unit tests

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update user guide

anchengyang committed:
Merge pull request #7 from anchengyang/branch-A-UserGuide

Update user guide

anchengyang committed:
Screenshot

anchengyang committed:
Update readme

anchengyang committed:
Update README.md

anchengyang committed:
Merge branch 'master' into branch-A-UserGuide

anchengyang committed:
Merge pull request #8 from anchengyang/branch-A-UserGuide

Update README.md

anchengyang committed:
Close stage

anchengyang committed:
checkstyle

anchengyang committed:
update user guide

anchengyang committed:
Update readme

anchengyang committed:
Update README.md

anchengyang committed:
Fix unamed module warning on app launch

anchengyang committed:
minor fixes

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Fix bug

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md

anchengyang committed:
Update README.md
